### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by geofjamg**
Signed-off-by: Geoffroy Jamgotchian <geoffroy.jamgotchian@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*
This is the initial commit of network map server web service that is tha back end part of the network map UI component.

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: src/main/java/com/powsybl/network/map/NetworkMapController.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+@RestController
+@RequestMapping(value = "/" + NetworkMapController.API_VERSION + "/")
+@Api(tags = "network-map-server")
+@ComponentScan(basePackageClasses = NetworkMapService.class)
+public class NetworkMapController {
+
+    public static final String API_VERSION = "v1";
+
+    @Inject
+    private NetworkMapService networkMapService;
```
Comment (AbdelHedhili) [inline src/main/java/com/powsybl/network/map/NetworkMapController.java]: Why do you use inject annotation and autowired. It doesn't change anything but it's probably best to only use one of them and not mix them up.

File: src/main/java/com/powsybl/network/map/NetworkMapService.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * @author Abdelsalem Hedhili <abdelsalem.hedhili at rte-france.com>
+ */
```
Comment (AbdelHedhili) [inline src/main/java/com/powsybl/network/map/NetworkMapService.java]: add yourself as author

File: .github/workflows/maven.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+      - name: Build with Maven
+        run: mvn --batch-mode package
+
+      - name: Run SonarCloud analysis
+        run: >
+          mvn --batch-mode -Pjacoco package sonar:sonar
+          -Dsonar.host.url=https://sonarcloud.io
+          -Dsonar.organization=powsybl-ci-github
+          -Dsonar.projectKey=com.powsybl:powsybl-network-map-server
```
Comment (AbdelHedhili) [inline .github/workflows/maven.yml]: shouldn't we add that part on another PR so it doesn't mark the build as failed ?

File: src/main/java/com/powsybl/network/map/NetworkMapController.java (hunk 1 above)
Comment (geofjamg) [reply src/main/java/com/powsybl/network/map/NetworkMapController.java]: Agreed, I replaced with @Autowired

File: src/main/java/com/powsybl/network/map/NetworkMapService.java (hunk 2 above)
Comment (geofjamg) [reply src/main/java/com/powsybl/network/map/NetworkMapService.java]: Fixed

File: .github/workflows/maven.yml (hunk 3 above)
Comment (geofjamg) [reply .github/workflows/maven.yml]: I found a solution to this problem. We just have to pre-create project on Sonar Cloud side before running the CI and there is no more error.

**Pull request merged**